cmd/infrakit/x/vmwscript.go

```diff
@@ -33,76 +31,38 @@ func vmwscriptCommand() *cobra.Command {
 			log.Crit("Please specify the path to a configuration file")
 			os.Exit(-1)
 		}
-		err := vmwscript.OpenFile(args[0])
+		plan, err := vmwscript.OpenFile(args[0])
 		if err != nil {
 			log.Crit("Error opening file", "Error", err)
 			os.Exit(-1)
 		}
 
-		// if configuration isn't set in JSON, check Environment vars/flags
-		if (vm.VCenterURL == nil) || *vm.VCenterURL == "" {
-			if vm.VCenterURL = vc; *vm.VCenterURL == "" {
-				log.Crit("VMware vCenter/vSphere credentials are missing")
-				os.Exit(-1)
-			}
-		}
-
-		if (vm.DCName == nil) || *vm.DCName == "" {
-			if vm.DCName = dc; *vm.DCName == "" {
-				log.Error("No Datacenter was specified, will try to use the default (will cause errors with Linked-Mode)")
-			}
-		}
-
-		if (vm.DSName == nil) || *vm.DSName == "" {
-			if vm.DSName = ds; *vm.DSName == "" {
-				log.Crit("A VMware vCenter datastore is required for provisioning")
-				os.Exit(-1)
-			}
-		}
-
-		if (vm.NetworkName == nil) || *vm.NetworkName == "" {
-			if vm.NetworkName = nn; *vm.NetworkName == "" {
-				log.Crit("Specify a Network to connect to")
-				os.Exit(-1)
-			}
-		}
-
-		if (vm.VSphereHost == nil) || *vm.VSphereHost == "" {
-			if vm.VSphereHost = vh; *vm.VSphereHost == "" {
-				log.Crit("A Host inside of vCenter/vSphere is required to provision on for VM capacity")
-				os.Exit(-1)
-			}
-		}
-
-		// Ideally these should be populated as they're needed for a lot of the tasks.
-		if (vm.VMTemplateAuth.Username == nil) || *vm.VMTemplateAuth.Username == "" {
-			if vm.VMTemplateAuth.Username = gu; *vm.VMTemplateAuth.Username == "" {
-				log.Error("No Username for inside of the Guest OS was specified, somethings may fail")
-			}
-		}
-
-		if (vm.VMTemplateAuth.Password == nil) || *vm.VMTemplateAuth.Password == "" {
-			if vm.VMTemplateAuth.Password = gp; *vm.VMTemplateAuth.Username == "" {
-				log.Error("No Password for inside of the Guest OS was specified, somethings may fail")
-			}
-		}
+		// set the values from flags
+		plan.VMWConfig.VCenterURL = vc
+		plan.VMWConfig.DCName = dc
+		plan.VMWConfig.DSName = ds
+		plan.VMWConfig.NetworkName = nn
+		plan.VMWConfig.VSphereHost = vh
+		plan.VMWConfig.VMTemplateAuth.Username = gu
+		plan.VMWConfig.VMTemplateAuth.Password = gp
```

Comment:
Originally we checked the `VMWConfig.{X}` to see if it had been set from the unmarshalling of the JSON, if not then we set it to what was either a flag or env variable. This change results in the wipe of any configuration set from the JSON to what is either a flag or environment variable.

Reply:
Ah sorry I missed that.  This should be easy to fix: we instantiate an empty Plan and the decode with the bytes from a file then call a Validate.  Will work on that right away.

Reply:
I meant after instantiating the Plan we set its fields with flags values and then Validate.